### PR TITLE
New authentication implementation

### PR DESCRIPTION
**Pull request by BullDog57Rus**

### CODE REVIEW

**Review by Zemidar — APPROVED** [review]

File: obs/obs-api.yaml [hunk 1]
```diff
@@ -120,20 +110,39 @@ paths:
...
+              schema:
+                type: string
+                format: binary
+        default:
+          description: Bad request, security violation, or internal server error.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseError'
+  /files/uploadFromStarter:
```
Comment (aehrenthal) [inline obs/obs-api.yaml:127]: I am not sure if this endpoint is perfectly named. I dont think the API should care whether or not the file comes from the starter library or in a different way. Can we maybe think of a better name to keep it more generic?

**Review by aehrenthal — CHANGES_REQUESTED** [review]

File: obs/obs-api.yaml (hunk 1 above)
Comment (BullDog57Rus) [reply obs/obs-api.yaml:127]: actually, it is a deprecated endpoint only to support talk roll from starter to microservice. We have to re-add all the files to the microservice

File: obs/obs-api.yaml [hunk 2]
```diff
@@ -120,20 +110,39 @@ paths:
...
+    post:
+      deprecated: true
+      summary: Readd the file uploaded by obs starter lib via the microservice
```
[the remark this answers is not in the document]
Comment (BullDog57Rus) [reply obs/obs-api.yaml:130]: re-add? it's like repeating addition :D

File: obs/obs-api.yaml [hunk 3]
```diff
@@ -168,10 +189,14 @@ components:
           type: string
         contentSize:
           description: The file size.
-          type: long
+          type: integer
+          format: int64
         description:
           description: The file description provides more information about the file. For example, why it has been uploaded, its author.
           type: string
+        publiclyAccessible:
```
Comment (aehrenthal) [inline obs/obs-api.yaml:197]: I think on the API level we can just do public as a boolean. In Spring it's a reserved keyword, so there it makes sense to use publiclyAccessible, but I think for the API it's nicer to just have a public boolean as we have for the other APIs such as admin/owner for the UserEntity.

**Review by aehrenthal — APPROVED** [review]

**Pull request merged**